soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -594,9 +540,10 @@ namespace Logic {
     MasterSword     = MasterSword   || IsAdult;
     BiggoronSword   = BiggoronSword || ProgressiveGiantKnife >= 2;
 
+    //you need at least 2 buttons for scarecrow song
     ScarecrowSong    = ScarecrowSong || FreeScarecrow || (ChildScarecrow && AdultScarecrow);
-    Scarecrow        = Hookshot && CanPlay(ScarecrowSong, Song::SONG_SCARECROW);
-    DistantScarecrow = Longshot && CanPlay(ScarecrowSong, Song::SONG_SCARECROW);
+    Scarecrow        = Hookshot && Ocarina && OcarinaButtons >= 2;
```

Comment:
Either account for FreeScarecrow here and in z_en_kakasi2.c, or clarify in the tooltip that scarecrow still needs 2 buttons. (I gave my intial comment on the setting assuming it would just be in logic with 0 buttons if free, but still requiring buttons and only saving the trip to lake is valid too)

Reply:
Updated the tooltip and hopefully made the logic easier to understand